digitalservicebund/ris-backend-service
# Create .env in ./run.sh + add instructions

How to test: 

- install direnv

```bash
brew install direnv
```

- add to the end of .zshrc (see [here](https://github.com/direnv/direnv/blob/master/docs/hook.md)):

```bash
eval "$(direnv hook zsh)"
```

- create/add to ~/.config/direnv/direnv.toml (see [here](https://github.com/direnv/direnv/blob/master/man/direnv.toml.1.md#codeloaddotenvcode))

```bash
[global]
load_dotenv = true
```

- in root (once and if values change)

```bash
./run.sh env
direnv allow .
```

- to test

```bash
echo $OAUTH2_CLIENT_ID
```

- to launch

```bash
./run.sh dev
```

- to launch from IDE, add the .env file in repo root. e.g. in vscode launch.json:

```bash
{
  "version": "0.2.0",
  "configurations": [
    {
      "type": "java",
      "name": "Launch Application",
      "request": "launch",
      "mainClass": "de.bund.digitalservice.ris.Application",
      "projectName": "ris-backend-service",
      "envFile": "${workspaceFolder}/../.env",
      "args": "--spring.profiles.active=local"
    }
  ]
}
```
